File: src/main/java/org/nia/logic/commands/FightClubCommands.java
package org.nia.logic.commands;

import org.nia.bots.CWTavernBot;
import org.nia.logic.ServingMessage;
import org.nia.model.TournamentUsers;
import org.nia.model.User;
import org.telegram.telegrambots.api.methods.send.SendMessage;
import org.telegram.telegrambots.api.objects.Message;
import org.telegram.telegrambots.api.objects.replykeyboard.InlineKeyboardMarkup;
import org.telegram.telegrambots.api.objects.replykeyboard.buttons.InlineKeyboardButton;
import org.telegram.telegrambots.exceptions.TelegramApiException;

import java.util.ArrayList;
import java.util.List;


/**
 * @author Иван, 11.03.2017.
 */
public enum FightClubCommands implements Commands {
    DRAKA("/DRAKA") {
        @Override
        public boolean isApplicable(Message message, User from) {
            if (!super.isApplicable(message, from)) {
                return false;
            }
            TournamentUsers currentByUserID = TournamentUsers.getCurrentByUserID(message.getFrom().getId());
            return currentByUserID != null && currentByUserID.isInFight() && currentByUserID.getScore() == 0;
        }

        @Override
        public String apply(Message message, User from) {
            TournamentUsers currentByUserID = TournamentUsers.getCurrentByUserID(message.getFrom().getId());
            User user = currentByUserID.getUser();
            currentByUserID.setScore(user.getFightClubStatsSum());
            currentByUserID.save();
            String res = String.format(currentByUserID.getTournament().getType().getStartPhrase(), user);
            if (message.isUserMessage()) {
                try {
                    SendMessage tournamentMessage = ServingMessage.getTournamentMessage(res + "\nКоличество болельщиков: 0");
                    setKeyboard(user, tournamentMessage);
                    CWTavernBot.INSTANCE.sendMessage(tournamentMessage);
                } catch (TelegramApiException e) {
                    e.printStackTrace();
                }
            } else {
                res += "\nКоличество болельщиков: 0";
            }

            return res;
        }
    };
    protected String text;

    FightClubCommands(String text) {
        this.text = text;
    }

    @Override
    public boolean isApplicable(Message message, User from) {
        return message.getText().contains(this.text);
    }

    @Override
    public SendMessage getMessage(Message message, String answer) {
        SendMessage sendMessage = new SendMessage();
        sendMessage.setChatId(message.getChat().getId());
        sendMessage.enableHtml(true);
        sendMessage.setText(answer);
        if (!message.isUserMessage()) {
            setKeyboard(User.getFromMessage(message.getFrom()), sendMessage);
        }

        return sendMessage;
    }

    protected void setKeyboard(User user, SendMessage sendMessage) {
        InlineKeyboardMarkup inlineKeyboardMarkup = new InlineKeyboardMarkup();
        List<List<InlineKeyboardButton>> keyboard = new ArrayList<>();
        List<InlineKeyboardButton> row = new ArrayList<>();
        InlineKeyboardButton button = new InlineKeyboardButton();
        button.setText("AVE_" + user.toString().replace("@", "").toUpperCase());
        button.setCallbackData(String.valueOf(user.getUserID()));
        row.add(button);
        keyboard.add(row);
        inlineKeyboardMarkup.setKeyboard(keyboard);
        sendMessage.setReplyMarkup(inlineKeyboardMarkup);
    }

    @Override
    public String apply(Message message, User from) {
        return "";
    }

}

File: src/main/java/ru/nia/tavern/service/commands/FightClubCommands.kt
package ru.nia.tavern.service.commands

import org.telegram.telegrambots.api.methods.send.SendMessage
import org.telegram.telegrambots.api.objects.Message
import org.telegram.telegrambots.api.objects.replykeyboard.InlineKeyboardMarkup
import org.telegram.telegrambots.api.objects.replykeyboard.buttons.InlineKeyboardButton
import org.telegram.telegrambots.exceptions.TelegramApiException
import ru.nia.tavern.model.TournamentUsers
import ru.nia.tavern.model.User
import ru.nia.tavern.service.CWTavernBot
import java.util.*

/**
 * @author Иван, 11.03.2017.
 */
enum class FightClubCommands private constructor(protected var text: String) : Commands {
    DRAKA("/DRAKA") {
        override fun isApplicable(message: Message, from: User): Boolean {
            if (!super.isApplicable(message, from)) {
                return false
            }
            val currentByUserID = TournamentUsers.getCurrentByUserID(message.from.id!!)
            return currentByUserID != null && currentByUserID.isInFight && currentByUserID.score == 0
        }

        override fun apply(message: Message, from: User): String {
            val currentByUserID = TournamentUsers.getCurrentByUserID(message.from.id!!)
            val user = currentByUserID!!.user
            currentByUserID.score = user.fightClubStatsSum
            currentByUserID.save()
            var res = String.format(currentByUserID.tournament.type.startPhrase, user)
            if (message.isUserMessage) {
                try {
                    val tournamentMessage = ServingMessage.getTournamentMessage("$res\nКоличество болельщиков: 0")
                    setKeyboard(user, tournamentMessage)
                    CWTavernBot.INSTANCE.sendMessage(tournamentMessage)
                } catch (e: TelegramApiException) {
                    e.printStackTrace()
                }

            } else {
                res += "\nКоличество болельщиков: 0"
            }

            return res
        }
    };

    override fun isApplicable(message: Message, from: User): Boolean {
        return message.text.contains(this.text)
    }

    override fun getMessage(message: Message, answer: String): SendMessage {
        val sendMessage = SendMessage()
        sendMessage.setChatId(message.chat.id!!)
        sendMessage.enableHtml(true)
        sendMessage.text = answer
        if (!message.isUserMessage) {
            setKeyboard(User.getFromMessage(message.from), sendMessage)
        }

        return sendMessage
    }

    protected fun setKeyboard(user: User, sendMessage: SendMessage) {
        val inlineKeyboardMarkup = InlineKeyboardMarkup()
        val keyboard = ArrayList<List<InlineKeyboardButton>>()
        val row = ArrayList<InlineKeyboardButton>()
        val button = InlineKeyboardButton()
        button.text = "AVE_" + user.toString().replace("@", "").toUpperCase()
        button.callbackData = user.userID.toString()
        row.add(button)
        keyboard.add(row)
        inlineKeyboardMarkup.keyboard = keyboard
        sendMessage.replyMarkup = inlineKeyboardMarkup
    }

    override fun apply(message: Message, from: User): String {
        return ""
    }

}
